### PR TITLE
Fix R

### PR DESCRIPTION
**Pull request by Thra11**
###### Motivation for this change
R was broken on aarch64 because
* It now seems to expect java to exist
* A recently added regression test fails on aarch64.

ZHF: #80379

###### Things done
*  Removed the conditional exclusion of jdk as a dependency on aarch64 (and aarch32), as aarch64 now has openjdk.
* Patched the sources to remove a test which fails on aarch64 (Reported upstream. Believe test to be incorrect).

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Thra11** [comment]
If preferred, we could only apply the patch on platforms where the test fails.

**Comment by xfix** [comment]
I don't know R, but I think it's a bug that on aarch64 there are no warnings. Asking for `2^64 %% 3` is nonsensical as at this point it's not possible to represent integers with double floating point numbers (you can only do up to 2^53 - 1).

I think it could make sense to limit disabling this test to aarch64.

**Comment by Thra11** [comment]
The calculation is done with long double on aarch64

**Comment by Thra11** [comment]
Although that's probably not relevant, since the value has already lost precision before the mod is calculated, so there should be a warning.

I do still think that it makes sense to suppress the test until it's fixed upstream. It's arguably no worse than the version before the test was added.

**Review by sugar700 — APPROVED** [review]

File: pkgs/applications/science/math/R/default.nix [hunk 1]
```diff
@@ -7,7 +7,6 @@
 # R as of writing does not support outputting both .so and .a files; it outputs:
 #     --enable-R-static-lib conflicts with --enable-R-shlib and will be ignored
 , static ? false
-, javaSupport ? (!stdenv.hostPlatform.isAarch32 && !stdenv.hostPlatform.isAarch64)
```
Comment (timokau) [inline pkgs/applications/science/math/R/default.nix]: Why did you remove this?

**Review by timokau — DISMISSED** [review]
Thanks for looking into this, especially for taking it upstream!

File: pkgs/applications/science/math/R/default.nix (hunk 1 above)
Comment (timokau) [reply pkgs/applications/science/math/R/default.nix]: Oh sorry, I missed that part of your PR description.

**Review by timokau — APPROVED** [review]
